### PR TITLE
Remove genereics & hard coded ValuesSource references from Matrix Stats

### PR DESCRIPTION
**Pull request by not-napoleon**
This is a relatively small PR to bring `ArrayValuesSourceAggregationBuilder` up to parity with `ValuesSourceAggregationBuilder` as far as the refactor goes.

### CODE REVIEW

File: modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregationBuilder.java [hunk 1]
```diff
@@ -260,76 +207,6 @@ protected abstract ArrayValuesSourceAggregatorFactory innerBuild(QueryShardConte
                                                                      AggregatorFactory parent,
                                                                      AggregatorFactories.Builder subFactoriesBuilder) throws IOException;
 
-    public ValuesSourceConfig config(QueryShardContext queryShardContext, String field, Script script) {
```
Comment (polyfractal) [inline modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregationBuilder.java]: Huh, was this all just copy pasta from the main VSConfig resolve?

**Review by polyfractal — APPROVED** [review]
LGTM

File: modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregationBuilder.java (hunk 1 above)
Comment (not-napoleon) [reply modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregationBuilder.java]: Not exactly.  There are differences, but I don't think they matter.  For example, this code ignores the script and just treats a query that specified a script as if it was unmapped.  Which, honestly, seems more like a bug waiting to happen than anything else.  (The parser already enforces not sending a script, so I'm not worried that going to suddenly start getting script errors)

Comment (polyfractal) [reply modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregationBuilder.java]: :+1:

**Comment by not-napoleon** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**